src/plugins/intel_cpu/src/snippets_transformations/enforce_precision.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets_transformations/enforce_precision.hpp"
+
+#include <assert.h>
+#include <memory>
+
+#include "ov_ops/type_relaxed.hpp"
+#include "snippets/itt.hpp"
+#include "ngraph/rt_info.hpp"
+#include "snippets/pass/propagate_precision.hpp"
+
+using namespace ngraph;
+using namespace ov::intel_cpu::pass;
+
+EnforcePrecision::EnforcePrecision(
+    const element::Type source,
+    const element::Type target,
+    const bool is_target_isa_supported,
+    std::function<EnforcePrecision::Operation(const std::shared_ptr<ngraph::Node>& op)> get_supported_precisions) :
+    source(source),
+    target(target),
+    is_target_isa_supported(is_target_isa_supported),
+    get_supported_precisions(get_supported_precisions == nullptr ? get_supported_precisions_default : get_supported_precisions) {
+}
+
+bool EnforcePrecision::run_on_model(const std::shared_ptr<ov::Model>& f) {
+    RUN_ON_MODEL_SCOPE(PropagatePrecision);
+    OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "ov::intel_cpu::pass::EnforcePrecision")
+
+    bool was_updated = false;
+    for (const auto& op : f->get_ordered_ops()) {
+        const auto& op_desc = get_supported_precisions(op);
+
+        if (op_desc.precisions.empty()) {
+            continue;
+        }
+
+        if (op_desc.require_target_isa && !is_target_isa_supported) {
+            continue;
+        }
+
+        std::vector<element::Type> actual_precisions;
+        for (const auto& input : op->inputs()) {
+            actual_precisions.push_back(input.get_element_type());
+        }
+
+        auto op_is_appropriate = true;
+
+        std::vector<element::Type> supported_precisions_to_enforce;
+        for (const auto& supported_precisions : op_desc.precisions) {
+            if (supported_precisions.size() != actual_precisions.size()) {
+                continue;
+            }
+
+            auto op_has_to_be_handled = false;
+            for (auto index = 0ull; index < supported_precisions.size(); ++index) {
+                if ((supported_precisions[index] == target) && (actual_precisions[index] == source)) {
+                    op_has_to_be_handled = true;
```

Comment:
thanks for comment, but it doesn't: different variables and different meaning

Reply:
Why so? Do we expect several identical instances in both `supported_precisions` and `actual_precisions`? So that `supported_precisions[i] == target && actual_precisions[i] == source` holds for two distinct `i1 and i2, where i1!=i2`?

Reply:
done. comments were added & variable was removed